### PR TITLE
bring over Labs updates

### PR DESCRIPTION
**Pull request by jstjoe**

### CODE REVIEW

**Comment by maximeprades** [comment]
@jstjoe @mrosendin what are we doing with this one?

File: app.js [hunk 1]
```diff
@@ -1,47 +1,84 @@
...
+// Initial Globals
+// this.settingsTags
+
 (function() {
 
   return {
 
     events: {
-      'app.activated':'initializeApp',
+      'app.created':'initializeApp',
```
Comment (jstjoe) [inline app.js]: these two prevent the app from running again each time an agent switches away from, and back to, a ticket

File: app.js [hunk 2]
```diff
@@ -1,47 +1,84 @@
...
+      'ticket.updated':'tagsLoaded',// this ensures other agents updating the ticket doesn't break anything
```
Comment (jstjoe) [inline app.js]: "this ensures other agents updating the ticket doesn't break anything" commented inline :)

File: app.js [hunk 3]
```diff
@@ -1,47 +1,84 @@
...
       'click .menu-toggle':'toggleMenu'
     },
+    requests: {
+        getRemoteTags: function() {
+            return {
+                url: this.setting("remoteTagsURI") || "https://jstjoe.github.io/json/remoteTags2.json",
+                cache: false
+            };
+        }
+    },
```
Comment (jstjoe) [inline app.js]: this is part of enabling the app to load the tags configuration remotely (to get around the DB limits on settings in apps)

**Comment by epritchard** [comment]
@jstjoe — Do you know what ever happened with this PR? Was it abandoned for good? I received a new bug report and want to try to get to that at some point, and just saw this outstanding PR.

**Comment by jstjoe** [comment]
The customer requesting the upgrade gave up on the app so I dropped it and then got shipped out to India. Do what you will with it :)

**Comment by epritchard** [comment]
Great, thanks for confirming @jstjoe. Hope all is well.

Cheers!

**Comment by epritchard** [comment]
Closing this for now. Will revisit later if we have more requests for remote tag storage.

**Pull request closed without merging**